Cargo.toml

```diff
@@ -47,13 +47,14 @@ miden-tx              = { default-features = false, path = "crates/miden-tx", ve
 miden-tx-batch-prover = { default-features = false, path = "crates/miden-tx-batch-prover", version = "0.11" }
 
 # Miden dependencies
-assembly       = { default-features = false, package = "miden-assembly", version = "0.15" }
-miden-crypto   = { default-features = false, version = "0.14" }
-miden-prover   = { default-features = false, version = "0.15" }
-miden-stdlib   = { default-features = false, version = "0.15" }
-miden-verifier = { default-features = false, version = "0.15" }
-vm-core        = { default-features = false, package = "miden-core", version = "0.15" }
-vm-processor   = { default-features = false, package = "miden-processor", version = "0.15" }
+assembly         = { default-features = false, package = "miden-assembly", version = "0.16.1" }
+miden-crypto     = { default-features = false, version = "0.15.5" }
+miden-prover     = { default-features = false, version = "0.16.1" }
+miden-stdlib     = { default-features = false, version = "0.16.1" }
+miden-utils-sync = { default-features = false, version = "0.16.1" }
+miden-verifier   = { default-features = false, version = "0.16.1" }
+vm-core          = { default-features = false, package = "miden-core", version = "0.16.1" }
+vm-processor     = { default-features = false, package = "miden-processor", version = "0.16.1" }
```

Comment:
I don't think we need to specify the patch versions here (can be changed in a subsequent PR).